tests/components/heos/test_config_flow.py

```diff
@@ -120,3 +120,59 @@ async def test_discovery_flow_aborts_already_setup(
 
     assert result["type"] is FlowResultType.ABORT
     assert result["reason"] == "single_instance_allowed"
+
+
+async def test_reconfigure_shows_form(hass: HomeAssistant, config_entry) -> None:
+    """Test reconfigure flow shows form with current host value."""
+    config_entry.add_to_hass(hass)
+
+    result = await config_entry.start_reconfigure_flow(hass)
+
+    host = next(
+        key.default() for key in result["data_schema"].schema if key == CONF_HOST
+    )
+    assert host == "127.0.0.1"
+    assert result["errors"] == {}
+    assert result["step_id"] == "reconfigure"
+    assert result["type"] is FlowResultType.FORM
+
+
+async def test_reconfigure_validates_and_updates_config(
+    hass: HomeAssistant, config_entry, controller
+) -> None:
+    """Test reconfigure validates host and successfully updates."""
+    config_entry.add_to_hass(hass)
+    result = await config_entry.start_reconfigure_flow(hass)
+    assert config_entry.data[CONF_HOST] == "127.0.0.1"
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        user_input={CONF_HOST: "127.0.0.2"},
+    )
+
+    assert controller.connect.call_count == 2  # Also called when entry reloaded
+    assert controller.disconnect.call_count == 1
+    assert config_entry.data == {CONF_HOST: "127.0.0.2"}
+    assert result["reason"] == "reconfigure_successful"
+    assert result["type"] is FlowResultType.ABORT
+
+
+async def test_reconfigure_cannot_connect_shows_form(
+    hass: HomeAssistant, config_entry, controller
+) -> None:
+    """Test reconfigure cannot connect and shows error form."""
+    controller.connect.side_effect = HeosError()
+    config_entry.add_to_hass(hass)
+    result = await config_entry.start_reconfigure_flow(hass)
+    assert config_entry.data[CONF_HOST] == "127.0.0.1"
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        user_input={CONF_HOST: "127.0.0.2"},
+    )
+
+    assert controller.connect.call_count == 1
+    assert controller.disconnect.call_count == 1
+    assert result["errors"][CONF_HOST] == "cannot_connect"
+    assert result["step_id"] == "reconfigure"
+    assert result["type"] is FlowResultType.FORM
```

Comment:
clear the error and bring the test to an `ABORT` state

Reply:
Can you clarify what you mean?

Reply:
we alway like to see that we can recover from errors for the config flows. So we  want to bring all tests to a terminal state `CREATE_ENTRY`/`ABORT`. So what I mean is: Remove the side effect and test you can now reconfigure successfully